### ISSUE TITLE
File is saved even if file transfer was cancelled

### ISSUE THREAD

**Issue by ovalseven8** [qTox/qTox#2490]
Hello,

when Alice is sending a file to Bob and the file transfer gets cancelled for whatever reason, the uncomplete and so corrupt file is saved anyway.

For me, it's a bug. I can't see any purpose why the file should be saved.

**Comment by agilob** [comment]
> file transfer gets cancelled

@zetok toxcore supports resuming file transfers right?

I would say, having a broken part of a file is better than not having anything after cancelled file transfer.

**Comment by ghost** [comment]
> better than not having anything

Unless the transfer was manually cancelled.

**Comment by SlugFiller** [comment]
I would actually view deleting the file as a bug. Especially if it's a particularly large file, and you spent some time getting those bytes. This is why, for example, torrent clients give you the option between deleting the torrent and data, to just deleting the torrent.

A lot of downloaders, including some browsers, don't view "Stop" as "Delete all the progress you have so far". In some cases (e.g. FTP), you can even choose to resume a file rather than overwrite, if you choose to save a new download to the same filename.

**Comment by ProMcTagonist** [comment]
This isn't a bug. File transfers are supposed to support resuming, even if that itself is buggy right now.

If collecting lots of abandoned .parts becomes a real concern, maybe clients could delete old ones after a set time period. Or just have a single Clean Up Old Canceled Files button in the Advanced settings.

**Comment by ovalseven8** [comment]
At least when the file transfer was manually cancelled, the file should be removed.

**Comment by TheNain38** [comment]
I think it would be better to delete the file if the receiver canceled the transfer, but if the sender canceled it I don't know which is better.

**Comment by ghost** [comment]
As long as it wasn't the receiver's intention to cancel the transfer, the partial download shouldn't be deleted.

**Comment by TheNain38** [comment]
@kehugter Yes, if the receiver canceled it, then it should delete it, but if the sender canceled it then it shouldn't. But file resuming won't work on that file if the sender resend it.

**Comment by ProMcTagonist** [comment]
If the sender resends it, he's resending it

**Comment by TheNain38** [comment]
@ProMcTagonist But there should be a way for the receiver to delete the file in the GUI, maybe I could try to partially save this issue in my PR about file transfers

**Comment by zetok** [comment]
> @zetok toxcore supports resuming file transfers right?

yep.

> But there should be a way for the receiver to delete the file in the GUI

Yep, I think that this should do instead of deleting something automagically, which might not be desired by user.

**Comment by TheNain38** [comment]
@zetok Even if the receiver stops the file transfer it shouldn't delete the file and then, in GUI, you could.
